lib/ParamedicSauceLabs.js

```diff
@@ -18,538 +18,553 @@
+                caps.tunnelIdentifier = process.env[utilities.SAUCE_TUNNEL_ID_ENV_VAR];
+            } else if (!caps.tunnelIdentifier) {
+                throw new Error('Testing browser platform on Sauce Labs requires Sauce Connect tunnel. Please specify tunnel identifier via --sauceTunnelId');
+            }
+            break;
+
+        default:
+            throw new Error('Don\'t know the Sauce caps for platform: ' + this.platformId);
+        }
 
-    // on platforms other than browser, only run sauce connect if user explicitly asks for it
-    if (!isBrowser && !self.config.getSauceTunnelId()) {
-        return Q();
+        return caps;
     }
-    // on browser, run sauce connect in any case
-    if (isBrowser && !self.config.getSauceTunnelId()) {
-        self.config.setSauceTunnelId(process.env[util.SAUCE_TUNNEL_ID_ENV_VAR] || self.config.getBuildName());
+
+    connectWebdriver () {
+        const user = this.config.getSauceUser();
+        const key = this.config.getSauceKey();
+        const caps = this.getSauceCaps();
+
+        logger.normal('cordova-paramedic: connecting webdriver');
+        const spamDots = setInterval(() => {
+            process.stdout.write('.');
+        }, 1000);
+
+        wd.configureHttp({
+            timeout: utilities.WD_TIMEOUT,
+            retryDelay: utilities.WD_RETRY_DELAY,
+            retries: utilities.WD_RETRIES
+        });
+
+        const driver = wd.promiseChainRemote(utilities.SAUCE_HOST, utilities.SAUCE_PORT, user, key);
+        return driver
+            .init(caps)
+            .then(() => {
+                clearInterval(spamDots);
+                process.stdout.write('\n');
+            }, (error) => {
+                clearInterval(spamDots);
+                process.stdout.write('\n');
+                throw (error);
+            });
     }
 
-    return Q.Promise(function (resolve, reject) {
-        logger.info('cordova-paramedic: Starting Sauce Connect...');
-        sauceConnectLauncher({
-            username: self.config.getSauceUser(),
-            accessKey: self.config.getSauceKey(),
-            tunnelIdentifier: self.config.getSauceTunnelId(),
-            connectRetries: util.SAUCE_CONNECT_CONNECTION_RETRIES,
-            connectRetryTimeout: util.SAUCE_CONNECT_CONNECTION_TIMEOUT,
-            downloadRetries: util.SAUCE_CONNECT_DOWNLOAD_RETRIES,
-            downloadRetryTimeout: util.SAUCE_CONNECT_DOWNLOAD_TIMEOUT
-        }, function (err, sauceConnectProcess) {
-            if (err) {
-                reject(err);
-            }
-            self.sauceConnectProcess = sauceConnectProcess;
-            logger.info('cordova-paramedic: Sauce Connect ready');
-            resolve();
+    connectSauceConnect () {
+        const isBrowser = this.isBrowser;
+
+        // on platforms other than browser, only run sauce connect if user explicitly asks for it
+        if (!isBrowser && !this.config.getSauceTunnelId()) return Q();
```

Comment:
Just a suggestion, but maybe this could be also changed to something like this.

```
if (!this.config.getSauceTunnelId()) {
  if (!isBrowser) {
    return Q();
 } else {
    this.config.setSauceTunnelId(process.env[utilities.SAUCE_TUNNEL_ID_ENV_VAR] || this.config.getBuildName());
  }
}
```

Reply:
This may lower the cyclomatic complexity but will look into apply this in a later PR.